packages/auxiliary-lines/src/AuxiliaryManager.ts

```diff
@@ -0,0 +1,362 @@
+      newPosition.z = 1;
+      newPosition.transformCoordinate(inverseProj);
+      localPositions.push(newPosition);
+    }
+
+    const indices = this.indices_;
+    indices.push(
+      OldPositionsLength,
+      1 + OldPositionsLength,
+      1 + OldPositionsLength,
+      2 + OldPositionsLength,
+      2 + OldPositionsLength,
+      3 + OldPositionsLength,
+      3 + OldPositionsLength,
+      OldPositionsLength, // front
+      OldPositionsLength,
+      4 + OldPositionsLength,
+      1 + OldPositionsLength,
+      5 + OldPositionsLength,
+      2 + OldPositionsLength,
+      6 + OldPositionsLength,
+      3 + OldPositionsLength,
+      7 + OldPositionsLength, // link
+      4 + OldPositionsLength,
+      5 + OldPositionsLength,
+      5 + OldPositionsLength,
+      6 + OldPositionsLength,
+      6 + OldPositionsLength,
+      7 + OldPositionsLength,
+      7 + OldPositionsLength,
+      4 + OldPositionsLength // back
+    );
+  }
+
+  /**
+   * Create auxiliary mesh for spot light.
+   * @param light - The SpotLight
+   */
+  addSpotLightAuxiliary(light: SpotLight) {
+    const transform = light.entity.transform;
+    this.transforms_.push(transform);
+    const height = light.distance;
+    const radius = Math.tan(light.angle) * height;
+
+    const localPositions = this.localPositions_;
+    const OldPositionsLength = localPositions.length;
+    this.transformRanges_.push(OldPositionsLength);
+    WireFramePrimitive.createConeWireFrame(radius, height, OldPositionsLength, localPositions, this.indices_);
+    this.isLocalDirty_ = true;
+    this.transformNoScaleFlag.push(true);
+  }
+
+  /**
+   * Create auxiliary mesh for point light.
+   * @param light - The PointLight
+   */
+  addPointLightAuxiliary(light: PointLight) {
+    const transform = light.entity.transform;
+    this.transforms_.push(transform);
+    const distance = light.distance;
+
+    const localPositions = this.localPositions_;
+    const OldPositionsLength = localPositions.length;
+    this.transformRanges_.push(OldPositionsLength);
+    WireFramePrimitive.createSphereWireFrame(distance, OldPositionsLength, localPositions, this.indices_);
+    this.isLocalDirty_ = true;
+    this.transformNoScaleFlag.push(true);
+  }
+
+  /**
+   * Create auxiliary mesh for directional light.
+   * @param light - The DirectLight
+   */
+  addDirectLightAuxiliary(light: DirectLight) {
+    const transform = light.entity.transform;
+    this.transforms_.push(transform);
+
+    const localPositions = this.localPositions_;
+    const OldPositionsLength = localPositions.length;
+    this.transformRanges_.push(OldPositionsLength);
+    WireFramePrimitive.createUnboundCylinderWireFrame(1, OldPositionsLength, localPositions, this.indices_);
+    this.isLocalDirty_ = true;
+    this.transformNoScaleFlag.push(true);
+  }
+
+  /**
+   * Create auxiliary mesh for box collider shape.
+   * @param shape - The BoxColliderShape
+   */
+  addBoxColliderShapeAuxiliary(shape: BoxColliderShape) {
```

Comment:
We can support two level method:
- Entity level: Entity (maybe)
- Compoment level: PointLight、Collider、Camera，all is Compoment 
- Property level: BoxColliderShape

So `addBoxColliderShapeAuxiliary ` is third level， you need another medthod `addColliderAuxiliary `( just some encapsulation)

Reply:
done